src/pages/workspace/card/issueNew/AssigneeStep.tsx

```diff
@@ -1,30 +1,91 @@
-import React from 'react';
+import React, {useMemo, useState} from 'react';
 import {View} from 'react-native';
-import FormProvider from '@components/Form/FormProvider';
+import type {OnyxEntry} from 'react-native-onyx';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import * as Expensicons from '@components/Icon/Expensicons';
 import InteractiveStepSubHeader from '@components/InteractiveStepSubHeader';
 import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import type {ListItem} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
 import Text from '@components/Text';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
+import {formatPhoneNumber} from '@libs/LocalePhoneNumber';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
 import Navigation from '@navigation/Navigation';
 import * as Card from '@userActions/Card';
 import CONST from '@src/CONST';
-import ONYXKEYS from '@src/ONYXKEYS';
+import type * as OnyxTypes from '@src/types/onyx';
 
-function AssigneeStep() {
+type AssigneeStepProps = {
+    // The policy that the card will be issued under
+    policy: OnyxEntry<OnyxTypes.Policy>;
+};
+
+function AssigneeStep({policy}: AssigneeStepProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
+    const [searchTerm, setSearchTerm] = useState('');
 
-    const submit = () => {
-        // TODO: the logic will be created in https://github.com/Expensify/App/issues/44309
-        Card.setIssueNewCardStep(CONST.EXPENSIFY_CARD.STEP.CARD_TYPE);
+    const submit = (assignee: ListItem) => {
+        Card.setIssueNewCardDataAndGoToStep({assigneeEmail: assignee?.login ?? ''}, CONST.EXPENSIFY_CARD.STEP.CARD_TYPE);
     };
 
     const handleBackButtonPress = () => {
         Navigation.goBack();
     };
 
+    const shouldShowSearchInput = policy?.employeeList && Object.keys(policy.employeeList).length > 8;
+    const textInputLabel = shouldShowSearchInput ? translate('workspace.card.issueNewCard.findMember') : undefined;
+
+    const membersDetails = useMemo(() => {
```

Comment:
For this, I think we need to add the sort function and filter function like we did in WorkspaceMembersPage

https://github.com/Expensify/App/blob/a84f68eeb1f72d0ca3308a4bdc9dc78e59a46d2e/src/pages/workspace/WorkspaceMembersPage.tsx#L324

https://github.com/Expensify/App/blob/a84f68eeb1f72d0ca3308a4bdc9dc78e59a46d2e/src/pages/workspace/WorkspaceMembersPage.tsx#L379

Reply:
Sorting is not required in the docs, but I can add it for consistency

Reply:
@koko57 Yeah, I suggest this for consistency. 

@mountiny Should we add the sort func in the member list here? I want to confirm that because it is outside our docs

Reply:
I think we can add the same logic as in the workspace members page yeah

Reply:
FYI: as isDeletedPolicyEmployee is used the same way in `WorkspaceMembersPage` and in the `AssigneeStep` I'm moving this to PolicyUtils